cordyceps/src/sorted_list.rs

```diff
@@ -0,0 +1,459 @@
+//! [Intrusive], singly-linked, sorted, linked-list.
+//!
+//! See the documentation for the [`SortedList`] and [`SortedListIter`] types for
+//! details.
+//!
+//! [intrusive]: crate#intrusive-data-structures
+
+use crate::{Linked, Stack};
+use core::{marker::PhantomData, ptr::NonNull};
+
+pub use crate::stack::Links;
+
+/// A sorted singly linked list
+///
+/// This behaves similar to [`Stack`], in that it is a singly linked list,
+/// however items are stored in an ordered fashion. This means that insertion
+/// is an `O(n)` operation, and retrieval of the first item is an `O(1)` operation.
+///
+/// It allows for a user selected ordering operation. If your type `T` implements
+/// [`Ord`]:
+///
+/// * Consider using [`SortedList::new_min()`] if you want **smallest** items sorted first.
+/// * Consider using [`SortedList::new_max()`] if you want **largest** items sorted first.
+///
+/// If your type `T` does NOT implement [`Ord`], or you want to use
+/// a custom sorting anyway, consider using [`SortedList::new_custom()`]
```

Comment:
Let's also add a note about the fact that elements in a `SortedList` must implement  `Linked<stack::Links<T>>`, and --- critically --- that this means that the SAME links are reused when an item is linked into the sorted list and when it's linked into a stack. That means that you can link the same element into either structure, but you can't have something that's linked into a sorted list and a stack at the same time (without wrapper structs that have separate sets of links, which is a thing we do in maitake a bit...)